span-processing-config-service-impl/src/main/java/org/hypertrace/span/processing/config/service/SpanProcessingConfigServiceImpl.java

```diff
@@ -116,9 +178,13 @@ public void updateExcludeSpanRule(
       this.validator.validateOrThrow(requestContext, request);
 
       UpdateExcludeSpanRule updateExcludeSpanRule = request.getRule();
+
+      // check if the rule already exists. If not, check if it is a system level config. If yes, use
+      // that. Else, error out.
       ExcludeSpanRule existingRule =
           this.excludeSpanRulesConfigStore
               .getData(requestContext, updateExcludeSpanRule.getId())
+              .or(() -> getSystemLevelExcludeSpanRule(updateExcludeSpanRule.getId()))
```

Comment:
How are we taking care of the fact, if a customer disables the system exclude rule, we need to store is as disabled?

Reply:
It is handled in building updated rule part on L189. The new rule would be built from the existing system exclude rule and the update exclude rule, and upserted.